sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_uamqp_transport.py

```diff
@@ -0,0 +1,605 @@
+            annotations=annotated_message.annotations,
+            delivery_annotations=annotated_message.delivery_annotations,
+            footer=annotated_message.footer
+        )
+
+    @classmethod
+    def create_named_key_token_credential(cls, credential):
+        return EventhubAzureNamedKeyTokenCredential(credential)
+
+    @classmethod
+    def create_shared_key_credential(cls, policy, key):
+        return EventHubSharedKeyCredential(policy, key)
+
+    def get_batch_message_encoded_size(self, message):
+        """
+        Gets the batch message encoded size given an underlying Message.
+        :param uamqp.BatchMessage message: Message to get encoded size of.
+        :rtype: int
+        """
+        return message.gather()[0].get_message_encoded_size()
+
+    def get_message_encoded_size(self, message):
+        """
+        Gets the message encoded size given an underlying Message.
+        :param uamqp.Message message: Message to get encoded size of.
+        :rtype: int
+        """
+        return message.get_message_encoded_size()
+
+    def get_remote_max_message_size(self, handler):
+        """
+        Returns max peer message size.
+        :param AMQPClient handler: Client to get remote max message size on link from.
+        :rtype: int
+        """
+        return handler.message_handler._link.peer_max_message_size  # pylint:disable=protected-access
+
+    def create_retry_policy(self, config):
+        """
+        Creates the error retry policy.
+        :param ~azure.eventhub._configuration.Configuration config: Configuration.
+        """
+        return errors.ErrorPolicy(max_retries=config.max_retries, on_error=_error_handler)
+
+    def create_link_properties(self, link_properties):
+        """
+        Creates and returns the link properties.
+        :param dict[bytes, int] link_properties: The dict of symbols and corresponding values.
+        :rtype: dict
+        """
+        return {types.AMQPSymbol(symbol): types.AMQPLong(value) for (symbol, value) in link_properties.items()}
+
+    def create_send_client(self, *, config, **kwargs): # pylint:disable=unused-argument
+        """
+        Creates and returns the uamqp SendClient.
+        :param ~azure.eventhub._configuration.Configuration config: The configuration.
+
+        :keyword str target: Required. The target.
+        :keyword JWTTokenAuth auth: Required.
+        :keyword int idle_timeout: Required.
+        :keyword network_trace: Required.
+        :keyword retry_policy: Required.
+        :keyword keep_alive_interval: Required.
+        :keyword str client_name: Required.
+        :keyword dict link_properties: Required.
+        :keyword properties: Required.
+        """
+        target = kwargs.pop("target")
+        retry_policy = kwargs.pop("retry_policy")
+        network_trace = kwargs.pop("network_trace")
+
+        return SendClient(
+            target,
+            debug=network_trace,  # pylint:disable=protected-access
```

Comment:
is this network_trace -- that might just be in pyamqp

Reply:
yep in uamqp it's debug and not network_trace. However, this reminds me that I need to update the pyamqp mgmt client creation so that we pass in network_trace instead of debug to AMQPClient .